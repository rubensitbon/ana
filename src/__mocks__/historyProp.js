const history = {
  length: 1,
  action: '',
  location: {
    pathname: '',
    search: '',
    hash: '',
    state: '',
  },
  push: jest.fn(),
  replace: jest.fn(),
  go: jest.fn(),
  goBack: jest.fn(),
  goForward: jest.fn(),
  block: jest.fn(),
};

export default history;
