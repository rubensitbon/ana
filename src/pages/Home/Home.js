// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import type { RouterHistory } from 'react-router-dom';

import StyledIntro from './Home.style';

const navigateTo = (history: RouterHistory): (() => void) => (): void => {
  history.push('/avatar');
};

type Props = {
  history: RouterHistory,
};

const Home = ({ history }: Props) => (
  <React.Fragment>
    <StyledIntro>
      <p>
        <FormattedMessage id="home.get-started" defaultMessage="To get started, edit" />
        <code className="intro-code">src/App.js</code>
        <FormattedMessage id="home.save-to-reload" />
      </p>
      <p>
        <FormattedMessage id="home.create-route" />
      </p>
      <p>
        <FormattedMessage id="home.generate-component" />
        <code className="intro-code">yarn generate</code>
      </p>
      <p>
        <FormattedMessage id="home.generate-module" />
      </p>
      <p>
        <FormattedMessage id="home.readme" />
      </p>

      <button onClick={navigateTo(history)}>
        <FormattedMessage id="home.click-me" />
      </button>
      <Link to="/avatar" href="/avatar">
        <FormattedMessage id="home.use-a-link" />
      </Link>
    </StyledIntro>
  </React.Fragment>
);

export default Home;
