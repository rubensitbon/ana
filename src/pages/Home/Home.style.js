import styled from 'styled-components';

const StyledIntro = styled.div`
  font-size: large;

  .intro-code {
    margin-left: 10px;
    margin-right: 10px;
  }
`;

export default StyledIntro;
