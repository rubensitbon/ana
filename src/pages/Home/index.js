// @flow
export { default } from './Home.wrap';
