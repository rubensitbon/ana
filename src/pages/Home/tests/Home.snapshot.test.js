// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import historyProp from '__mocks__/historyProp';
import Home from '../Home';

describe('<Home>', () => {
  it('should render correctly', () => {
    const props = { history: historyProp };

    const tree = shallow(<Home {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
