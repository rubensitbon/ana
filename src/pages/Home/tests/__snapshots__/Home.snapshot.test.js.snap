// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> should render correctly 1`] = `
<React.Fragment>
  <styled.div>
    <p>
      <FormattedMessage
        defaultMessage="To get started, edit"
        id="home.get-started"
        values={Object {}}
      />
      <code
        className="intro-code"
      >
        src/App.js
      </code>
      <FormattedMessage
        id="home.save-to-reload"
        values={Object {}}
      />
    </p>
    <p>
      <FormattedMessage
        id="home.create-route"
        values={Object {}}
      />
    </p>
    <p>
      <FormattedMessage
        id="home.generate-component"
        values={Object {}}
      />
      <code
        className="intro-code"
      >
        yarn generate
      </code>
    </p>
    <p>
      <FormattedMessage
        id="home.generate-module"
        values={Object {}}
      />
    </p>
    <p>
      <FormattedMessage
        id="home.readme"
        values={Object {}}
      />
    </p>
    <button
      onClick={[Function]}
    >
      <FormattedMessage
        id="home.click-me"
        values={Object {}}
      />
    </button>
    <Link
      href="/avatar"
      replace={false}
      to="/avatar"
    >
      <FormattedMessage
        id="home.use-a-link"
        values={Object {}}
      />
    </Link>
  </styled.div>
</React.Fragment>
`;
